### PR TITLE
LPS-114468 Fix keyword generation for options

### PR DESCRIPTION
**Pull request by carolmariaabb**

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js [hunk 1]
```diff
@@ -197,6 +197,9 @@ const Options = ({
 				id
 			);
 		}
+		else if (property === 'value') {
+			value = normalizeValue(fields, value, fields[index]);
+		}
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js:202]: @carolmariaabb, please re-use the previous `if` for optimization, something like:
```
if (property === 'value') {
    if (generateKeyword) {
        ...
    }
    else {
        ...
    }
}
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js [hunk 2]
```diff
@@ -316,21 +313,15 @@ const Main = ({
...
+				{({defaultOptionRef, handleField, index, option}) => (
 					<KeyValue
 						generateKeyword={option.generateKeyword}
 						keyword={option.value}
 						keywordReadOnly={
 							defaultLanguageId !== editingLanguageId
 						}
 						name={`option${index}`}
-						onBlur={handleBlur}
+						onBlur={() => {}}
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js:324]: @carolmariaabb, if there are no more handlers, what it's the purpose of keeping the prop `onBlur`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js [hunk 3]
```diff
@@ -340,7 +331,7 @@ const Main = ({
 								defaultOptionRef.current = false;
 							}
 						}}
-						onKeywordBlur={handleBlur}
+						onKeywordBlur={() => {}}
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js:334]: @carolmariaabb, if there are no more handlers, what it's the purpose of keeping the prop `onKeywordBlur`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/util.es.js [hunk 4]
```diff
@@ -105,9 +105,8 @@ export const dedupValue = (fields, value, id) => {
  * 1. If the current value is null, use the label
  * 2. If the current label is null, use the string Option
  */
-export const normalizeValue = (fields, currentField) => {
-	const {label, value: prevValue} = currentField;
-	let value = prevValue ? prevValue : label;
+export const normalizeValue = (fields, currentValue, currentField) => {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/util.es.js:108]: @carolmariaabb, can we add `currentValue` to be after `currentField` due to the alphabetic order?

**Pull request closed without merging**